check.go

```diff
@@ -57,6 +57,70 @@ func (d *DBFT[H]) checkCommit() {
 		return
 	}
 
+	// TODO: it should be considered: add PreCommit message instead of CommitAck and
+	// send this PreCommit *before* Commit. This will allow to create a completely
+	// custom message for keys exchange whereas the final Commit will keep only final
+	// signature bytes, as with usual dBFT.
+	if d.isAntiMEVExtensionEnabled() {
+		d.preBlock = d.CreatePreBlock()
+		hash := d.preBlock.Hash()
+
+		d.Logger.Info("processing PreBlock",
+			zap.Uint32("height", d.BlockIndex),
+			zap.Stringer("preBlock hash", hash),
+			zap.Int("tx_count", len(d.preBlock.Transactions())))
+
+		d.preBlockProcessed = true
+		d.ProcessPreBlock(d.preBlock)
+
+		if d.CommitSent() { // TODO: Do we really need to require Commit sent by *self* or M other's Commits is enough to sent CommitAck? It depends on the keys sharing logic
```

Comment:
The only questionable scenario I can think of is: you don't get enough PrepareResponses to make a Commit yourself, but you get M Commits. In this case it should be possible for you to construct an appropriate block and send a CommitAck. But this goes against the scheme, we never had a PrepareResponse->CommitAck transition there, so for now it'd be a bit safer to rely on `CommitSent()`.

Reply:
Good, let's keep it. TODO is removed, comment is added and now it's `PreCommitSent()`.